game/effect.go

```diff
@@ -1,8 +1,49 @@
+/*
+	An Effect can be created by a spell, a triggered ability, or an
+	activated ability.
+
+	An Effect can be the property of a Card. For example, when the
+	Card is a Giant Growth, its effect property would be
+	&Effect{power:3, toughness:3}.
+
+	An Effect can also be the Morbid or Kicker property of a Card, to designate a
+	Effect that only happens under special circumstances.
+
+*/
+
 package game
 
 import ()
 
 type Effect struct {
-	Action *Action
-	Card   *Card
+
+	// from Card
+	CastingCost        *CastingCost // when an Effect is a kicker, it has a Cost
+	Hexproof           bool
+	Kicker             *Effect
+	Plus1Plus1Counters int
+	Power              int
+	Summon             *Card
+	Toughness          int
+	Untargetable       bool
+
+	// from Action
```

Comment:
these fields don't seem to make sense, why does an effect have a `With`, or a `WithKicker`, etc

Reply:
1) You actually don't need the WithKicker or WithPhyrexian, I'll get rid of those.

2) I think you do need the With - you need to know what permanent originated an Effect because some Effects depend on the state of the Permanent on resolution.

Reply:
"With" isn't what permanent originated the effect, though. Is it? Why would that be called "with". I thought "with" was just for attacking and blocking.

Reply:
Oh, I started using it for mana actions too, like saccing the eldrazi token. I can change that.

Reply:
"With" is probably not a good name of that field on the action eh. Well, as long as Effect is commented so that I could just read this file and figure out what the fields mean, it is decent.